packages/compiler-core/src/transforms/transformElement.ts

```diff
@@ -927,3 +929,12 @@ function stringifyDynamicPropNames(props: string[]): string {
 function isComponentTag(tag: string) {
   return tag === 'component' || tag === 'Component'
 }
+
+function hasMultipleChildren(node: ComponentNode | PlainElementNode): boolean {
+  // filter out potential comment nodes.
+  const children = (node.children = node.children.filter(
+    c => c.type !== NodeTypes.COMMENT 
+  ))
+
+  return children.length !== 1
```

Comment:
Shouldn't this be `children.length > 0`? If there are no children, `hasMultipleChildren()` would return `true`, which seems odd in itself, and that would cause a compiler error in the flow above.

Reply:
in line **149** it checks if `node.children.length > 0` , so If there are no children it wont get to `hasMultipleChildren()`, and won't cause a compiler error, but yet i'm not sure it should be `>1` or `!==1` because it doesn't really make a deference, what do you think?

but the compiler does causes a weird error if the only child is a comment though, same thing for Transition. which i think is not that important.

Reply:
Oh, gotcha. See my suggestion in the other comment, which would obviate `hasMultipleChildren()` here.